mlflow/sagemaker/container/__init__.py

```diff
@@ -66,9 +66,9 @@ def _serve():
         print("activating custom environment")
         env = conf[pyfunc.ENV]
         env_path_dst = os.path.join("/opt/mlflow/", env)
-        # check that the mlflow version is matching
-        if not os.path.isdir("/opt/mlflow"):
-            os.mkdir("/opt/mlflow")
+        env_path_dst_dir = os.path.dirname(env_path_dst)
```

Comment:
It's insufficient to create the top-level `/opt/mlflow` directory if `env` is a nested directory. We need to create the entire tree.

Reply:
Good catch